File: mobile/androidApp/src/main/java/com/vanessa/ui/screens/mood/MoodScreen.kt
package com.vanessa.ui.screens.mood

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.vanessa.core.domain.MoodType
import com.vanessa.core.ui.ThemeTokens
import com.vanessa.core.ui.colorHex
import com.vanessa.di.AppContainer

@Composable
fun MoodScreen(container: AppContainer, onDismiss: () -> Unit) {
    var picked by remember { mutableStateOf<MoodType?>(null) }
    var impulsive by remember { mutableStateOf(false) }
    Box(
        Modifier
            .fillMaxSize()
            .background(Color(ThemeTokens.palette.background))
            .padding(24.dp)
    ) {
        Column(Modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally) {
            Spacer(Modifier.height(40.dp))
            Text("Como você está agora?",
                color = Color(ThemeTokens.palette.textPrimary),
                style = MaterialTheme.typography.headlineSmall,
                fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.height(8.dp))
            Text("Sua emoção influencia suas decisões financeiras.",
                color = Color(ThemeTokens.palette.textSecondary))
            Spacer(Modifier.height(32.dp))

            MoodType.entries.forEach { mood ->
                Box(
                    Modifier
                        .fillMaxWidth()
                        .padding(vertical = 6.dp)
                        .clip(RoundedCornerShape(16.dp))
                        .background(Color(ThemeTokens.palette.surface))
                        .clickable {
                            picked = mood
                            val result = container.moodVm.pickMood(mood)
                            impulsive = result.isImpulsive
                        }
                        .padding(16.dp)
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Box(
                            Modifier
                                .size(40.dp)
                                .clip(RoundedCornerShape(20.dp))
                                .background(Color(mood.colorHex()).copy(alpha = 0.2f)),
                            contentAlignment = Alignment.Center
                        ) {
                            Text(when (mood) {
                                MoodType.ANSIEDADE -> "🧠"
                                MoodType.TEDIO -> "😐"
                                MoodType.EUFORIA -> "✨"
                                MoodType.TRISTEZA -> "🌧"
                                MoodType.CALMARIA -> "🍃"
                            }, color = Color(mood.colorHex()))
                        }
                        Spacer(Modifier.width(12.dp))
                        Text(mood.label, color = Color(ThemeTokens.palette.textPrimary),
                            style = MaterialTheme.typography.titleMedium)
                    }
                }
            }

            if (picked != null) {
                Spacer(Modifier.height(16.dp))
                if (impulsive) {
                    Surface(
                        color = Color(ThemeTokens.palette.warning),
                        shape = RoundedCornerShape(16.dp),
                        modifier = Modifier.fillMaxWidth()
                    ) {
                        Column(Modifier.padding(16.dp)) {
                            Text("⚠ Pausa",
                                color = Color.Black,
                                fontWeight = FontWeight.Bold)
                            Text("Alguns gastos impulsivos acontecem quando estamos ${picked!!.label.lowercase()}. " +
                                    "Antes de registrar uma despesa nova, considere esperar 30 min.",
                                color = Color.Black)
                        }
                    }
                }
                Spacer(Modifier.height(24.dp))
                Button(onClick = onDismiss) { Text("Continuar") }
            }
        }
    }
}

File: mobile/iosApp/Vanessa/Screens/MoodScreen.swift
import Foundation
import SwiftUI
import VanessaShared
import shared

struct MoodScreen: View {
    let vm: MoodViewModel
    @State private var picked: MoodType? = nil
    let onClose: () -> Void

    init(vm: MoodViewModel, onClose: @escaping () -> Void) {
        self.vm = vm
        self.onClose = onClose
    }

    var body: some View {
        VStack(spacing: 16) {
            Text("Como você está agora?")
                .font(.title2).foregroundColor(ThemePalette.textPrimary).bold()
            Text("Sua emoção influencia suas decisões financeiras.")
                .foregroundColor(ThemePalette.textSecondary)

            ForEach(MoodType.allTypes, id: \.self) { mood in
                Button(action: {
                    picked = mood
                    let res = vm.pickMood(mood: mood)
                    picked = mood
                    _ = res.isImpulsive
                }) {
                    HStack {
                        Text(mood.icon).font(.title)
                        Text(mood.label).foregroundColor(ThemePalette.textPrimary)
                        Spacer()
                    }
                    .padding(16)
                    .background(ThemePalette.surface.cornerRadius(12))
                }
            }

            if let picked = picked, picked.isImpulsive {
                VStack(alignment: .leading) {
                    Text("⚠ Pausa").bold()
                    Text("Alguns gastos impulsivos acontecem quando estamos \(picked.label.lowercased()). Antes de registrar uma despesa nova, considere esperar 30 min.")
                }
                .padding()
                .background(ThemePalette.warning.cornerRadius(12))
            }
            Button("Continuar") { onClose() }
                .buttonStyle(.borderedProminent)
            Spacer()
        }
        .padding(20)
    }
}

extension MoodType {
    static var allTypes: [MoodType] { [.ansiedade, .tedio, .euforia, .tristeza, .calmaria] }
}
